target/linux/lantiq/dts/FRITZ7312.dts

```diff
@@ -0,0 +1,140 @@
+/dts-v1/;
+
+#include "ar9.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "FRITZ7312 - 1&1 WLAN-MODEM";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+
+		led-internet = &info_green;
+		led-dsl = &power;
+		led-wifi = &wlan;
+	};
+
+	memory@0 {
+		reg = <0x0 0x4000000>;
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x1000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					ath9k_cal: partition@0 {
+						label = "urlader";
+						reg = <0x00000 0x20000>;
+						read-only;
+					};
+
+					partition@20000 {
+						label = "firmware";
+						reg = <0x20000 0xf60000>;
+					};
+
+					partition@f80000 {
+						label = "tffs (1)";
+						reg = <0xf80000 0x40000>;
+						read-only;
+					};
+
+					partition@fc0000 {
+						label = "tffs (2)";
+						reg = <0xfc0000 0x40000>;
+						read-only;
+					};
+				};
+			};
+		};
+
+		gpio: pinmux@E100B10 {
+			pinctrl-names = "default";
+			pinctrl-0 = <&state_default>;
+
+			state_default: pinmux {
+			};
+		};
+
+		etop@E180000 {
+			mtd-mac-address = <&ath9k_cal 0xa91>;
+			mtd-mac-address-increment = <(-2)>;
+		};
+
+		ifxhcd@E101000 {
+			status = "okay";
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <100>;
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+		dect {
+			label = "dect";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_PHONE>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		power: power {
+			label = "fritz7320:green:power";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+			default-state = "keep";
+		};
+		voice {
+			label = "fritz7320:green:fon";
+			gpios = <&gpio 47 GPIO_ACTIVE_LOW>;
+		};
+		dect {
+			label = "fritz7320:green:dect";
+			gpios = <&gpio 38 GPIO_ACTIVE_LOW>;
+		};
+		wlan: wlan {
+			label = "fritz7320:green:wlan";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+		info_green: info_green {
```

Comment:
After some testing found a BUG?

-> tried to set eth0 (single lan port) as WAN uplink (dhcp-client mode) -> eth0 is up but gets no IP-address (got one or 2 cases it got an IP after playing around for a while, could not reproduce it, no internet connectivity)

common case:
-> if eth0 (lan port) is bridged to wlan (stock lede configuration: static address, dhcp-server running) eth0 is up and IPs are assigned to clients both on lan and wlan

-> have both Lan&Wan to be bridged?
- wifi is working if put from bridged LANinterface to an other interface serving as dhcp-server,
- eth0 (single LanPort) gets no ip in dhcp-client mode (both in LANinterface or another f.e. WAN),
**in syslog i can see uhcpd sending dhcp discoveries, but gets no lease** -> connected to an other router (dhcpsever) or cable modem (dhcpserver)
- also tried vlan tagging aka: eth0.1 (port1 lan) and eth0.2 (port 2 wifi)

-> swconfig shows no switch, is there an internal ar9 one?

-> also played around with commits from https://github.com/xdarklight/openwrt/commits/fb7312_wip_20160114, comparing and adding them to https://git.lede-project.org/?p=lede/mkresin/staging.git;a=summary
- did not change behavior - would leave them out, "less is more"

-> think its somewhere in software: dhcp? routing?